### ISSUE TITLE
Change from '.. otherwise make using this software.' to '...write...'.

### ISSUE THREAD

**Issue by makkus** [licensezero/parity-public-license#15]
Hi @kemitchell,

I just re-read the Parity license and realised it might not be a good fit for the project I was intending to license with it after all. Then I figured out you made some changes since the last time I read it, from:

> 3. Release all source code for software that you analyze, change, or otherwise make using this software.

to

> 3. Release all source code for software whose source code you write, change, or analyze with this software.

Would you mind sharing your thinking behind this? The git commit says 'Focus rule 3 on manipulation of source code', which I take to mean that Parity should only apply to source code, not 'software' in the larger sense. 

My use-case is a build tool that doesn't necessarily manipulate or touch source code. Before that change, if you for example use it to build a Docker image containing non-free software (binaries) it'd have been a valid argument (in my view) to not allow this kind of usage, as you made a new piece of software (the Docker container/image). In the new version my reading is that this would be allowed (because non of the requirements write/change/analyze were hit), as long as one shares the Dockerfile (configuration) that builds the container containing proprietary software.

It might very well be that my use-case isn't one that should be covered by Parity, which would be fair enough. But maybe not? I could always use the older version of the License, or Prosperity. But might be good to clarify this point?
￼

**Comment by makkus** [comment]
Thanks @kemitchell, that makes sense. The project I'm thinking about is a more or less complete rewrite of this: https://docs.freckles.io/en/latest/overview.html

I haven't gotten around to figure out a good way to explain what it does in a sentence or two, but in a nutshell it helps you set up machines (virtual or physical) in a way so it can run your project/service. Basically configuration management, but with a few twists, and (hopefully) without the learning curve.

In order to do that it needs some sort of metadata, which can be either a configuration file or it can try to parse your project folder to figure out the nature of the project itself (the latter might fall in the 'analyze' category, although it doesn't really analyze source code itself, but a folder tree).

This probably falls somewhere between the 'normal' Parity use case, and the one with the Kanban board you used as an example. The source code of the project itself is not really relevant (except for the one case where it might be analyzed) for my project to do its thing (it could just use already compiled binaries), but it still works with the 'software' of the project, as opposed to the Kanban case where the software/source code is never touched at all.

So, my reading of the current license text is that using it on closed-source projects wouldn't trigger any restrictions, you might have to release the configuration files necessary, but nothing else. Ideally, I'd like the license I use to require the project and its dependencies to all be licensed 'on terms as permissive' as the Parity license. If that is not the case, there is not really a lot incentive for people using my project to either open source what they are working on, or getting a private license. On the other hand, maybe I'm overreaching a bit here. Pipelines/applications that call my program still need to be licensed permissively, the same applies if anybody wanted to build on top of it. Am I making any sense?

To be honest, I'm not sure myself whether this is a class of problem that is relevant enough for Parity to change its wording. The earlier version would have been a fairly good fit for me, mostly because it doesn't specify 'source code' and has the 'otherwise make' wording. But I can see that it leaves a bit too much room for interpretation. On the other hand, the current wording might ignore a whole category of tools surrounding the development process of software that still feel like they should be covered by a license like Parity. I guess the question is whether Parity should focus on covering source code, or software in general. There's a case to be made for both options.

This would probably also work for me:

> Release all source code for software you build, change, or analyze using this software.

'Build' can be read two ways though, in the 'compiling/assembling' sense, and the one that is very close to the 'make' from the earlier version again, a bit closer on the 'write'-side though, I think.

Those are things I personally would want to be encompassed by the 'making' section of the text:

- write source code/software project files
- change source code/software project files
- analyzes a software project or its files
- build/compile software
- deploy software
- configure software

Basically, whenever the Parity licensed project 'touches' or 'looks at' another software (or its source code). Again, I'm not sure whether that is not a bit too far reaching.

**Comment by makkus** [comment]
Another quick thought, maybe it helps to look at it from the point of the outcome: if the outcome is software, then the relevant parts involved in building/making it (that can have a license attached to it -- I don't think configuration files can/should be covered) should be licensed using a free license. 

I'm not suggesting to change the wording that radically. Just as an exercise, looking at it from a different perspective can maybe help clarifying relevant characteristics.

**Comment by kemitchell** [comment]
@makkus I think you're right. Focusing on software output is exactly what I've wanted to do. I don't mean to play coy lawyer, but the question then is "what does software mean".

"Making software" is very broad. "Making source code" leaves out a bunch of concrete things that might be "software", like configuration, documentation, or even design documents.

I think the current language is too narrow. I need to find a way to expand it without catching cases that nobody seems to think reasonable right now.
